<div class="alert alert-info" role="alert">
<p class ="lead"> 
<h1>Tiro Parabólico </h1>
<h2> Computación </h2>
<h2>Física, Facultad de Ciencias. UNAM </h2>
<h3>Celeste Castro Granados</h3>
</div>


Lo primero es definir el sistema de ecuaciones que necesitamos.

Sistema de ecuaciones que rigen el tiro parabólico:

$$
\textbf{v} = v_0 cos(\theta)\hat{i} + v_0 sen(\theta)\hat{j}
$$

$$
\textbf{a} = -g\hat{j}
$$

$$
\textbf{r} = (v_{0x}t+x_0)\hat{i} + \left(-\frac{1}{2}gt^2+v_{0y}t+y_0\right)\hat{j}
$$

El usuario puede elegir el tiempo, la velocidad inicial y como es un tiro parabólico el ángulo de tiro con el que quiera trabajar. En este caso elegimos arbitrariamente, pero siempre se le puede asignar un nuevo valor al anterior.

Entonces, elegimos primero un tiempo:

In [1]:
t = collect(0:0.125:20);

La constante que siempre vamos a tener es la gravedad. Que en el caso de la Tierra es ≈ 9.81 m/s²

In [2]:
g= 9.81 # m/s²

9.81

Asignamos nuestra velocidad inicial: 

In [3]:
V_0 = 100 #m/s

100

Y un ángulo de tiro.

In [4]:
θ = π/4

0.7853981633974483

Empezamos a definir la velocidad:

In [5]:
V = V_0*cos(θ)+V_0*sin(θ) #Con la ecuación planteada arriba

141.4213562373095

Y la aceleración:

In [6]:
a = -g

-9.81

Si nuestras condiciones iniciales son X=0 y Y=0:

$$
\textbf{r} = (v_{0x}t)\hat{i} + \left(-\frac{1}{2}gt^2+v_{0y}t\right)\hat{j}
$$

Vamos a tener una posición r para cada tiempo, es decir, partimos de ceros:

In [7]:
X = zeros(t);
Y = zeros(t);

Aquí hacemos una prueba para ver cual es, por ejemplo, el valor 132 dentro de nuestro intervalo de tiempo.

In [8]:
t[132]

16.375

Y ahora si escribimos el programa para obtener las posiciones X,Y:

In [9]:
for i in eachindex(t)
    X[i] = V_0*cos(θ)*t[i] #Como i va a ir cambiado, va a haber una posición X,Y para cada i
    Y[i] = (-(1/2*g*t[i]^2)+V_0*sin(θ)*t[i])
end

    

In [10]:
using Plots

In [11]:
pyplot()

Plots.PyPlotBackend()

In [12]:
gr()

Plots.GRBackend()

Ahora vamos a graficar este tiro parabólico:

In [13]:
plot(X,Y,w=2)
title!("Tiro parabólico")
xaxis!("Posición en x (m)")
yaxis!("Posición en y (m)")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0 
 
 
 250 
 
 
 500 
 
 
 750 
 
 
 1000 
 
 
 1250 
 
 
 -400 
 
 
 -200 
 
 
 0 
 
 
 200 
 
 
 Tiro parabólico 
 
 
 Posición en x (m) 
 
 
 Posición en y (m) 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

También podemos hacer un gif del tiro:

In [14]:
plt = plot(1,xlim=(0,1500), ylim=(-500,300), xlab="Posición en x(m)",ylab="Posición en y(m)", title="Tiro Parabólico")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0 
 
 
 500 
 
 
 1000 
 
 
 1500 
 
 
 -400 
 
 
 -200 
 
 
 0 
 
 
 200 
 
 
 Tiro Parabólico 
 
 
 Posición en x(m) 
 
 
 Posición en y(m) 
 
 
 
 
 
 y1

In [15]:
@gif for i in eachindex(t)
    x = V_0*cos(θ)*t[i]
    y = (-(1/2*g*t[i]^2)+V_0*sin(θ)*t[i])
    push!(plt, x, y)
    end every 2


INFO: Saved animation to /home/juser/compufisica2017/Notebooks_Celeste/tmp.gif


Plots.AnimatedGif("/home/juser/compufisica2017/Notebooks_Celeste/tmp.gif")

In [16]:
using Interact 

Y en este caso, vamos a graficar como cambian la posición X,Y variando el ángulo de tiro:

In [17]:
@manipulate for θ=0:π/16:2π
    X = V_0*cos(θ).*t
    Y = (-(0.5*g.*t.^2)+V_0*sin(θ).*t)
    plot(X,Y,w=2,xlim=(minimum(X),maximum(X)),ylim=(minimum(Y),maximum(Y)))
    title!("Tiro parabólico")
    xaxis!("Posición en X")
    yaxis!("Posición en Y")
end


Interact.Options{:SelectionSlider,Any}(1: "input" = 3.141592653589793 Any , "θ", 3.141592653589793, "3.141592653589793", 17, Interact.OptionDict(DataStructures.OrderedDict{Any,Any}("0.0"=>0.0,"0.19634954084936207"=>0.19635,"0.39269908169872414"=>0.392699,"0.5890486225480862"=>0.589049,"0.7853981633974483"=>0.785398,"0.9817477042468103"=>0.981748,"1.1780972450961724"=>1.1781,"1.3744467859455345"=>1.37445,"1.5707963267948966"=>1.5708,"1.7671458676442586"=>1.76715…), Dict{Any,Any}(Pair{Any,Any}(0.785398, "0.7853981633974483"),Pair{Any,Any}(5.69414, "5.6941366846315"),Pair{Any,Any}(5.89049, "5.890486225480862"),Pair{Any,Any}(0.589049, "0.5890486225480862"),Pair{Any,Any}(5.49779, "5.497787143782138"),Pair{Any,Any}(1.76715, "1.7671458676442586"),Pair{Any,Any}(2.35619, "2.356194490192345"),Pair{Any,Any}(1.37445, "1.3744467859455345"),Pair{Any,Any}(0.0, "0.0"),Pair{Any,Any}(0.392699, "0.39269908169872414")…)), Any[], Any[], true, "horizontal")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 -2000 
 
 
 -1500 
 
 
 -1000 
 
 
 -500 
 
 
 0 
 
 
 -1500 
 
 
 -1000 
 
 
 -500 
 
 
 0 
 
 
 Tiro parabólico 
 
 
 Posición en X 
 
 
 Posición en Y 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

**Nota: El punto es para multiplicar entrada a entrada**

# Tiempo de vuelo 

Tenemos la siguiente ecuación de posición, dada en sus coordenadas x,y:

$$
\textbf{r} = (v_{0x}t)\hat{i} + \left(-\frac{1}{2}gt^2+v_{0y}t\right)\hat{j}
$$

El tiempo de vuelo es el tiempo total del movimiento, por lo tanto para calcularlo tenemos que tomar en cuenta que Y = 0, que es cuando la partícula llega al suelo, tenemos por consiguiente la ecuación de la posición en Y: 

$$
\textbf{y} = \left(-\frac{1}{2}gt^2+v_{0y}t\right)\hat{j}
$$



Ahora como Y = 0

$$
\textbf{0} = -\frac{1}{2}gt^2+v_{0y}t
$$

Dividiendo entre t tenemos:

$$
\textbf{0} = -\frac{1}{2}gt+v_{0y}
$$

Y de esta ecuación despejamos el tiempo: 

$$
\textbf{t} = \frac{2v_{0y}}{g}
$$

Por consiguiente:

$$
\textbf{t_vuelo} = \frac{2v_{0y}}{g}
$$

In [18]:
Voy = V_0*sin(θ) #m/s

70.71067811865474

In [19]:
g #m/s^2

9.81

Entonces tenemos: 

In [20]:
t_vuelo = (2*Voy)/g #s

14.416040391163046

El tiempo de vuelo de la partícula va a ser de 14.416040391163046 segundos.

# Altura máxima

Cuando la partícula alcance su punto máximo, es decir su posición en y máxima, su velocidad se volverá cero, por lo demás tenemos los mismos datos de aceleración y velocidad inicial en y:

In [21]:
V = 0 #m/s

0

In [22]:
a #m/s^2

-9.81

In [23]:
Voy #m/s

70.71067811865474

Entonces la Y máxima va a ser:

$$
\textbf{Ymáx} = \frac{V^2 - Voy^2}{2a}
$$

Donde V = 0 m/s

In [24]:
Ymáx = V^2-Voy^2/2a #m

254.8419979612639

Por lo tanto la altura máxima de la partícula va a ser de 254.8419979612639 m

# Alcance máximo

Tenemos la siguiente ecuación de posición, dada en sus coordenadas x,y:

$$
\textbf{r} = (v_{0x}t)\hat{i} + \left(-\frac{1}{2}gt^2+v_{0y}t\right)\hat{j}
$$


Por lo tanto, para el alcance máximo nos va a interesar su coordenada en X, tenemos la ecuación de X:

$$
\textbf{x} = (v_{0x}t)\hat{i}
$$


Para este caso, tomaremos como t el tiempo de vuelo, ya que con ese tiempo tendremos la x máxima a la que llega la partícula

Tenemos el mismo dato de velocidad inicial en x:

In [25]:
Vox = V_0*cos(θ) #m/s

70.71067811865476

Ahora si calculamos el alcance máximo:

In [26]:
Xmáx = Vox*t_vuelo #m

1019.3679918450559

Otra manera para calcular el alcance máximo, sería sustituyendo la ecuación del tiempo que obtuvimos para el tiempo de vuelo en la ecuación de X de la ecuación de posición, y simplificando; lo cual nos arroja la siguiente ecuación:

$$
\textbf{Xmáx} = \frac{2v_0^2 sen (\theta) cos (\theta)}{g}
$$


Observemos que está ecuación es más general.
Si la probamos, nos da el mismo resultado para Xmáx:

In [27]:
Xmáx = 2*V_0^2*sin(θ)*cos(θ) / g

1019.367991845056

Por lo tanto, el alcance máximo de la partícula va a ser de 1019.3679918450559 m

## Ahora, ya que hemos calculado tiempo de vuelo, altura y alcance máximo, para los datos anteriores, podemos empezar a graficar de manera general variando, por ejemplo, el ángulo de disparo y la velocidad inicial.

En esta gráfica se puede observar la variación de la posición X,Y, el alcance máximo, altura máxima y tiempo de vuelo; mientras manipulamos el ángulo de tiro y la velocidad inicial.

In [28]:
@manipulate for θ=0:π/16:π/2, V_0=0:10:200 
    X = V_0*cos(θ).*t
    Y = (-(1/2*g.*t.^2)+V_0*sin(θ).*t)
    
    t_vuelo = (round(2*V_0*sin(θ)/g))
    Xmáx = (round(2*V_0^2*sin(θ)*cos(θ)/g))
    Ymáx = (round((-(V_0*sin(θ))^2)/2a))

    plot(X,Y,w=2,xlim=(minimum(X),maximum(X)),ylim=(minimum(Y),maximum(Y)))
    title!("Tiro parabólico")
    xaxis!("Posición en X(m)       Alcance = $Xmáx   m         Tiempo de vuelo = $t_vuelo    s")
    yaxis!("Posición en Y(m)       Altura máxima = $Ymáx  m")
end


Interact.Options{:SelectionSlider,Any}(5: "input-2" = 0.7853981633974483 Any , "θ", 0.7853981633974483, "0.7853981633974483", 5, Interact.OptionDict(DataStructures.OrderedDict{Any,Any}("0.0"=>0.0,"0.19634954084936207"=>0.19635,"0.39269908169872414"=>0.392699,"0.5890486225480862"=>0.589049,"0.7853981633974483"=>0.785398,"0.9817477042468103"=>0.981748,"1.1780972450961724"=>1.1781,"1.3744467859455345"=>1.37445,"1.5707963267948966"=>1.5708), Dict{Any,Any}(Pair{Any,Any}(0.0, "0.0"),Pair{Any,Any}(0.392699, "0.39269908169872414"),Pair{Any,Any}(0.785398, "0.7853981633974483"),Pair{Any,Any}(1.5708, "1.5707963267948966"),Pair{Any,Any}(0.589049, "0.5890486225480862"),Pair{Any,Any}(0.981748, "0.9817477042468103"),Pair{Any,Any}(0.19635, "0.19634954084936207"),Pair{Any,Any}(1.1781, "1.1780972450961724"),Pair{Any,Any}(1.37445, "1.3744467859455345"))), Any[], Any[], true, "horizontal")

Interact.Options{:SelectionSlider,Any}(7: "input-3" = 100 Any , "V_0", 100, "100", 11, Interact.OptionDict(DataStructures.OrderedDict{Any,Any}("0"=>0,"10"=>10,"20"=>20,"30"=>30,"40"=>40,"50"=>50,"60"=>60,"70"=>70,"80"=>80,"90"=>90…), Dict{Any,Any}(Pair{Any,Any}(200, "200"),Pair{Any,Any}(100, "100"),Pair{Any,Any}(50, "50"),Pair{Any,Any}(80, "80"),Pair{Any,Any}(40, "40"),Pair{Any,Any}(140, "140"),Pair{Any,Any}(190, "190"),Pair{Any,Any}(0, "0"),Pair{Any,Any}(160, "160"),Pair{Any,Any}(170, "170")…)), Any[], Any[], true, "horizontal")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0 
 
 
 250 
 
 
 500 
 
 
 750 
 
 
 1000 
 
 
 1250 
 
 
 -400 
 
 
 -200 
 
 
 0 
 
 
 200 
 
 
 Tiro parabólico 
 
 
 Posición en X(m) Alcance = 1019.0 m Tiempo de vuelo = 14.0 s 
 
 
 Posición en Y(m) Altura máxima = 255.0 m 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

En la siguiente gráfica se puede observar como cambia el alcance máximo dependiendo del ángulo de tiro, manipulando la velocidad inicial.

In [29]:
@manipulate for V_0=0:10:200
   # X = V_0*cos(θ).*t
   # Y = (-(1/2*g.*t.^2)+V_0*sin(θ).*t)
    
    θ2 = collect(0:π/64:π/2)
   # t_vuelo = (round(2*V_0*sin(θ) / g))
    Xmáx = (2*V_0^2.*sin.(θ2).*cos.(θ2) / g )
    plot(θ2,Xmáx,w=2)
    
    title!("Tiro parabólico")
    xaxis!("Ángulo de disparo (radianes)")
    yaxis!("Alcance(m)")
end

Interact.Options{:SelectionSlider,Any}(11: "input-4" = 100 Any , "V_0", 100, "100", 11, Interact.OptionDict(DataStructures.OrderedDict{Any,Any}("0"=>0,"10"=>10,"20"=>20,"30"=>30,"40"=>40,"50"=>50,"60"=>60,"70"=>70,"80"=>80,"90"=>90…), Dict{Any,Any}(Pair{Any,Any}(200, "200"),Pair{Any,Any}(100, "100"),Pair{Any,Any}(50, "50"),Pair{Any,Any}(80, "80"),Pair{Any,Any}(40, "40"),Pair{Any,Any}(140, "140"),Pair{Any,Any}(190, "190"),Pair{Any,Any}(0, "0"),Pair{Any,Any}(160, "160"),Pair{Any,Any}(170, "170")…)), Any[], Any[], true, "horizontal")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 0 
 
 
 250 
 
 
 500 
 
 
 750 
 
 
 1000 
 
 
 Tiro parabólico 
 
 
 Ángulo de disparo (radianes) 
 
 
 Alcance(m) 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

En la siguiente, se observa como cambia la altura máxima dependiendo del ángulo de tiro y variando la velocidad inicial.

In [30]:
@manipulate for V_0=0:10:200
    
    θ4=collect(0:π/32:π/2)
    Ymáx = ((-(V_0*sin.(θ4)).^2)/2a)
    
    
    plot(θ4,Ymáx,w=2)
    title!("Tiro parabólico")
    xaxis!("Ángulo de disparo (radianes)")
    yaxis!("Altura(m)")
end

Interact.Options{:SelectionSlider,Any}(15: "input-5" = 100 Any , "V_0", 100, "100", 11, Interact.OptionDict(DataStructures.OrderedDict{Any,Any}("0"=>0,"10"=>10,"20"=>20,"30"=>30,"40"=>40,"50"=>50,"60"=>60,"70"=>70,"80"=>80,"90"=>90…), Dict{Any,Any}(Pair{Any,Any}(200, "200"),Pair{Any,Any}(100, "100"),Pair{Any,Any}(50, "50"),Pair{Any,Any}(80, "80"),Pair{Any,Any}(40, "40"),Pair{Any,Any}(140, "140"),Pair{Any,Any}(190, "190"),Pair{Any,Any}(0, "0"),Pair{Any,Any}(160, "160"),Pair{Any,Any}(170, "170")…)), Any[], Any[], true, "horizontal")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 0 
 
 
 100 
 
 
 200 
 
 
 300 
 
 
 400 
 
 
 500 
 
 
 Tiro parabólico 
 
 
 Ángulo de disparo (radianes) 
 
 
 Altura(m) 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

Y finalmente en la siguiente gráfica se observa como cambia el tiempo de vuelo dependiendo del ángulo de disparo, manipulando la velocidad inicial.

In [31]:
@manipulate for V_0=0:10:200
    θ3=collect(0:π/32:π/2)
    t_vuelo = ((2*V_0.*sin.(θ3))/g)
    plot(θ3,t_vuelo,w=2)
    title!("Tiro parabólico")
    xaxis!("Ángulo de disparo (radianes)")
    yaxis!("Tiempo de vuelo(s)")
end
    

Interact.Options{:SelectionSlider,Any}(19: "input-6" = 100 Any , "V_0", 100, "100", 11, Interact.OptionDict(DataStructures.OrderedDict{Any,Any}("0"=>0,"10"=>10,"20"=>20,"30"=>30,"40"=>40,"50"=>50,"60"=>60,"70"=>70,"80"=>80,"90"=>90…), Dict{Any,Any}(Pair{Any,Any}(200, "200"),Pair{Any,Any}(100, "100"),Pair{Any,Any}(50, "50"),Pair{Any,Any}(80, "80"),Pair{Any,Any}(40, "40"),Pair{Any,Any}(140, "140"),Pair{Any,Any}(190, "190"),Pair{Any,Any}(0, "0"),Pair{Any,Any}(160, "160"),Pair{Any,Any}(170, "170")…)), Any[], Any[], true, "horizontal")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 0 
 
 
 5 
 
 
 10 
 
 
 15 
 
 
 20 
 
 
 Tiro parabólico 
 
 
 Ángulo de disparo (radianes) 
 
 
 Tiempo de vuelo(s) 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1